addon/components/labs-ui/legend-icon.js

```diff
@@ -1,19 +1,8 @@
-import Component from '@ember/component';
-import { computed } from '@ember/object';
-import layout from '../../templates/components/labs-ui/legend-icon';
+import Component from '@glimmer/component';
 
-export default Component.extend({
-  init(...args) {
-    this._super(...args);
-  },
-
-  icon: null,
-
-  classNames: ['legend-icon'],
-  layout,
-
-  iconType: computed(function() {
-    const type = this.get('icon.type');
+export default class LegendIconComponent extends Component {
+  get iconType() {
+    const type = this.args.iconType;
```

Comment:
should this be `... = this.args.icon.type`?

Will the passed in value of `this.args.icon.type`/`this.args.iconType` ever change?

Reply:
No, it won't change. Or, rather, we've not dynamically changed legend items before.